mailpoet/lib/Automation/Integrations/WooCommerce/Fields/CustomerOrderFieldsFactory.php

```diff
@@ -0,0 +1,217 @@
+        __('Average spent', 'mailpoet'),
+        function (CustomerPayload $payload) {
+          $customer = $payload->getCustomer();
+          $totalSpent = $customer ? (float)$customer->get_total_spent() : 0.0;
+          $orderCount = $customer ? (int)$customer->get_order_count() : 0;
+          return $orderCount > 0 ? ($totalSpent / $orderCount) : 0.0;
+        }
+      ),
+      new Field(
+        'woocommerce:customer:order-count',
+        Field::TYPE_INTEGER,
+        __('Order count', 'mailpoet'),
+        function (CustomerPayload $payload) {
+          $customer = $payload->getCustomer();
+          return $customer ? $customer->get_order_count() : 0;
+        }
+      ),
+      new Field(
+        'woocommerce:customer:first-paid-order-date',
+        Field::TYPE_DATETIME,
+        __('First paid order date', 'mailpoet'),
+        function (CustomerPayload $payload) {
+          $customer = $payload->getCustomer();
+          if (!$customer) {
+            return null;
+          }
+          return $this->getPaidOrderDate($customer, true);
+        }
+      ),
+      new Field(
+        'woocommerce:customer:last-paid-order-date',
+        Field::TYPE_DATETIME,
+        __('Last paid order date', 'mailpoet'),
+        function (CustomerPayload $payload) {
+          $customer = $payload->getCustomer();
+          if (!$customer) {
+            return null;
+          }
+          return $this->getPaidOrderDate($customer, false);
+        }
+      ),
+      new Field(
+        'woocommerce:customer:purchased-categories',
+        Field::TYPE_ENUM_ARRAY,
+        __('Purchased categories', 'mailpoet'),
+        function (CustomerPayload $payload) {
+          $customer = $payload->getCustomer();
+          return $customer ? $this->getOrderProductTermIds($customer, 'product_cat') : [];
+        },
+        [
+          'options' => $this->getTermsList('product_cat'),
+        ]
+      ),
+      new Field(
+        'woocommerce:customer:purchased-tags',
+        Field::TYPE_ENUM_ARRAY,
+        __('Purchased tags', 'mailpoet'),
+        function (CustomerPayload $payload) {
+          $customer = $payload->getCustomer();
+          return $customer ? $this->getOrderProductTermIds($customer, 'product_tag') : [];
+        },
+        [
+          'options' => $this->getTermsList('product_tag'),
+        ]
+      ),
+    ];
+  }
+
+  private function getPaidOrderDate(WC_Customer $customer, bool $fetchFirst): ?DateTimeImmutable {
+    $wpdb = $this->wordPress->getWpdb();
+    $sorting = $fetchFirst ? 'ASC' : 'DESC';
+    $statuses = array_map(function (string $status) {
+      return "wc-$status";
+    }, $this->wooCommerce->wcGetIsPaidStatuses());
+    $statusesPlaceholder = implode(',', array_fill(0, count($statuses), '%s'));
+
+    if ($this->wooCommerce->isWooCommerceCustomOrdersTableEnabled()) {
+      $statement = (string)$wpdb->prepare("
+        SELECT o.date_created_gmt
+        FROM {$wpdb->prefix}wc_orders o
```

Comment:
Not sure if HPOS plays into this. Orders used to be a post type, I remember we had something like a switch whether HPOS is active or not?